### ISSUE TITLE
Doesn't fetch custom git repo from Cargo.toml 

### ISSUE THREAD

**Issue by Corallus-Caninus** [rust-analyzer/rust-analyzer#9748]
when I change a crate to my fork using a github https repo RustAnalyzer seems to not recognize my feature additions (but still recognizes the original crates features) leading me to believe it's not fetching my repo after the change to Cargo.toml.

Thank you Rust Analyzer is awesome!

**Comment by lnicola** [comment]
Does it work after you restart the server (or editor)?

There's an issue causing us to only process the first `Cargo.toml` change that happens outside the editor.

**Comment by Corallus-Caninus** [comment]
Yes this is possible after a reboot the addition is found but new additions recur the same problem.

**Comment by lnicola** [comment]
How are you adding the new dependencies? By editing `Cargo.toml`, or with `cargo-edit`?

**Comment by Corallus-Caninus** [comment]
I am using cargo-edit

**Comment by Corallus-Caninus** [comment]
Can you or anyone else please reference the issue so I may investigate?

**Comment by lnicola** [comment]
It might be https://github.com/rust-analyzer/rust-analyzer/issues/9546.

**Comment by ChayimFriedman2** [comment]
Probably irrelevant by now.
